Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UCIRangeClauseFilter.cs

```diff
@@ -0,0 +1,932 @@
+                {
+                    tRanges.Add(new Tuple<T, T>(ConvertToContainedGeneric(ral.Item1), ConvertToContainedGeneric(ral.Item2)));
+                }
+
+                _ranges.Clear();
+                foreach (var key in allRanges.Keys)
+                {
+                    if (tRanges.Contains(allRanges[key]))
+                    {
+                        _ranges.Add(allRanges[key]);
+                        tRanges.Remove(allRanges[key]);
+                    }
+                }
+                _ranges.AddRange(tRanges); //what's left is the concatenated result
+            }
+        }
+
+        private int RangesValuesCount()
+        {
+            int result = 0;
+            foreach (var range in _ranges)
+            {
+                result = result + (int)(ToLong(range.Item2) - ToLong(range.Item1) + 1);
+            }
+            return result;
+        }
+
+        private T ConvertToContainedGeneric<K>(K value)
+        {
+            var uciVal = _valueFactory.Create(value.ToString(), TypeName);
+            return _tConverter(uciVal);
+        }
+
+        private bool TryMergeWithOverlappingRange(T start, T end)
+        {
+            var endIsWithin = _ranges.Where(t => t.Item1.CompareTo(end) < 0 && t.Item2.CompareTo(end) > 0);
+            var startIsWithin = _ranges.Where(t => t.Item1.CompareTo(start) < 0 && t.Item2.CompareTo(start) > 0);
+
+            var rangeIsAdded = false;
+            if (endIsWithin.Any() || startIsWithin.Any())
+            {
+                if (endIsWithin.Any())
+                {
+                    var original = endIsWithin.First();
+                    _ranges.Remove(endIsWithin.First());
+                    _ranges.Add(new Tuple<T, T>(start, original.Item2));
+                    rangeIsAdded = true;
+                }
+                else
+                {
+                    var original = startIsWithin.First();
+                    _ranges.Remove(startIsWithin.First());
+                    _ranges.Add(new Tuple<T, T>(original.Item1, end));
+                    rangeIsAdded = true;
+                }
+            }
+
+            return rangeIsAdded;
+        }
+
+        private void RemoveIsClauseImpl(string opSymbol)
+        {
+            if (_isClause.Keys.Contains(opSymbol))
+            {
+                if (_hasExtents)
+                {
+                    _isClause.Remove(opSymbol);
+                    var extentVal = opSymbol.Equals(CompareTokens.LT) ? _minExtent : _maxExtent;
+                    AddIsClauseImpl(extentVal, opSymbol);
+                }
+                else
+                {
+                    _isClause.Remove(opSymbol);
+                }
+            }
+        }
+
+        private string GetSinglesDescriptor()
+        {
+            var singles = _singleValues.Select(sv => sv.ToString()).ToList();
+            singles.AddRange(_variableSingles);
+            return GetSingleValueTypeDescriptor(singles, "Single=");
```

Comment:
It's not obvious from this code but is this ultimately going to be user-facing? If so, then we need to make use of `RubberduckUI` to support i18n for the data type description.

Reply:
The `ToString()` is not intended to ever be user-facing.  It is never presented to the user in the current implementation.